pkg/core/config/authentication.go

```diff
@@ -4,7 +4,7 @@ package config
 type AuthenticationConfiguration struct {
 	Identities                  []string                    `json:"identities,omitempty" yaml:"identities" msg:"identities"`
 	PrimaryAuthenticators       []string                    `json:"primary_authenticators,omitempty" yaml:"primary_authenticators" msg:"primary_authenticators"`
-	SecondaryAuthenticators     []string                    `json:"secondary_authenticators,omitempty" yaml:"secondary_authenticators" msg:"secondary_authenticators"`
+	SecondaryAuthenticators     *[]string                   `json:"secondary_authenticators,omitempty" yaml:"secondary_authenticators" msg:"secondary_authenticators"`
```

Comment:
We have 2 nil cases here.

The pointer to `[]string` is nil.
The pointer to `[]string` is non-nil but `[]string` is nil.

Do we really need to change the type here? For example `LoginIDKeys` is still `[]LoginIDKeyConfiguration` but we explicitly check it against nil to see if it is specified by the developer.

Reply:
I'll change it back.

Go's `nil` handling is really confusing... It is convention to treat `nil` slice and empty slice as same, yet encoding tends to treat them differently.

Reply:
golang-encoding impedance mismatch